### ISSUE TITLE
VS Extension: There was an error activating the remote language server, CodeStream.

### ISSUE THREAD

**Issue by xsoul** [TeamCodeStream/CodeStream#81]
After update to version 1.3.4 for VS Extension, opening a solution in VS 2019 will have that error. CodeStream panel always shows "Open a solution to start using CodeStream" although the solution already opened.
![err1](https://user-images.githubusercontent.com/5644701/62206642-da5cfd80-b3c4-11e9-8abf-4167640de9d5.PNG)

**Comment by bcanzanella** [comment]
Hi @xsoul sorry you're hitting this issue, but we'd like to get to the bottom of it... 

Please email me at brian@codestream.com these few things:

1. From Visual Studio, go to `Help -> About Microsoft Visual Studio` then click `Copy Info` and paste that into the email.

2. Attach vs-extension.log and vs-agent.log from `%localappdata%\CodeStream\Logs` as well as the ActivityLog.xml from `%AppData%\Microsoft\VisualStudio\16.0_<RandomText>` and attach them to the email.

**Comment by bcanzanella** [comment]
thanks @xsoul I received your email. I will dig into this and let you know if I need anything else. thank you!

**Comment by bcanzanella** [comment]
Hi @xsoul

The error you're seeing (below) _appears_ related to Visual Studio, in that it thinks your Visual Studio version doesn't support CodeActions (which it should).

```
2019-07-31 18:09:29.628 [15544] FATL [01] LanguageClient            OnServerInitializeFailedAsync
System.MissingMethodException: Method not found: 'Void Microsoft.VisualStudio.LanguageServer.Protocol.TextDocumentClientCapabilities.set_CodeAction(Microsoft.VisualStudio.LanguageServer.Protocol.DynamicRegistrationSetting)'.
   at Microsoft.VisualStudio.LanguageServer.Client.RemoteLanguageClientInstance.<InitializeAsync>d__52.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.VisualStudio.LanguageServer.Client.RemoteLanguageClientInstance.InitializeAsync(String workspaceLocation, CancellationToken token)
   at Microsoft.VisualStudio.LanguageServer.Client.RemoteLanguageClientInstance.<ActivateLanguageClientAsync>d__50.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
 ```

I've tried reporoducing this, using the same version of Visual Studio as you (16.0.28803.352), as well as the same list of installed extensions and there isn't an error for me. A few questions:

1) Have you recently installed, removed, or upgraded any .NET framework or Visual Studio component?

2) If there was one, what was the file extension of the active document in the editor when you opened your solution? What types of projects exist in the solution you're opening? (c#, f#, asp.net mvc, etc.). Does this error happen with all solutions you open?

3) Could you report back the versions of Microsoft.VisualStudio.LanguageServer.Client & Protocol dlls that are installed on your machine? You can run the following Powershell commands to get that info: (it's possible they have the same version)

 ```
 [System.Diagnostics.FileVersionInfo]::GetVersionInfo("C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\PublicAssemblies\Microsoft.VisualStudio.LanguageServer.Client.dll").FileVersion
```

```
[System.Diagnostics.FileVersionInfo]::GetVersionInfo("C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\PublicAssemblies\Microsoft.VisualStudio.LanguageServer.Protocol.dll").FileVersion
```

 There's a more detailed set of logs that VS can create when there are activation issues (the error infobar from your screenshot mentions it, i think). If you run the following Powershell command to open Visual Studio:

 `& "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\devenv.exe" /Log`

 ...and wait for the error to happen, the ActivityLog.xml should have some additional info `%AppData%\Microsoft\VisualStudio\16.0_<RandomText>\ActivityLog.xml`. Again, please email me that file at brian@codestream.com

**Comment by bcanzanella** [comment]
Hi @xsoul I wanted to give you an update.

I was finally able to recreate the issue that you are seeing, though oddly enough after installing the same version of Visual Studio 2019 as you I was only able to trigger the issue with Visual Studio 2017 and not in Visual Studio 2019.

The error, it seems, is not completely caused by the CodeStream extension, as once I started getting the error with CodeStream, I also got it with the LanguageServer sample in VSSDK-Extensibility-Samples (https://github.com/microsoft/VSSDK-Extensibility-Samples/tree/master/LanguageServerProtocol) -- which leads me to think that it's an issue either with the Visual Studio installation or the LanguageServer assemblies that are shipped with your version of Visual Studio. Essentially VS is in a state that may prohibit any extension that uses the Microsoft LanguageServer client from working.

To fix this, I had to uninstall and reinstall all versions of Visual Studio (including the build tools).
